### DISCUSSION TITLE
UI Stack

### DISCUSSION THREAD

**Discussion by michaelfortunato** [free2pee/free2pee#11] [Ideas]
We have decided to "opt-out" of managing a database for mvp (P's bday) and instead query open street maps for all of our bathroom questions.
Now comes the issue of choosing which frontend technology to use. 
The first task we have is to render an HTML table with one column, where each record contains a hyperlink to the next nearest bathroom on google maps. On clicking that link, the user is brought into the google maps app and directed to the restroom. 

There have been two proposals so far

# 1. Oxygen Julia
This is a server side rendering approach where an initial HTTP get query is made to our frontend and Julia will make the API request to OSM to render the table. This rendering only happens once and no http requests are made after the rendered HTML is delivered to the users browser

# 2. Nextjs
We will do the same server side rendering approach but instead our technology is nextjs. Nextjs is _the_ entireprise react framework and allows for server side rendering and client side rendering of html. It is more powerlful and flexble and I am familiar with it. 

I vote for nextjs.

**Comment by michaelfortunato** [comment]
@acoh64 This won't impact any ML initiative you have started. This is orthogonal to your department's efforts.

**Comment by michaelfortunato** [comment]
@ilanhaskel  Please let us know when you are planning to take PTO. It is company policy to give a two weeks notice but we understand if that can't happen. Just try to get it into workday so it's on payroll's records. 
Best,
Michael 
Distinguished Kernel Dev @ Free2Pee (F2P) 
We're hiring!

**Comment by anandijain** [comment]
I think there is a lot of value in keeping the quantity of technologies that we have to manage limited. For now, I propose we keep the fat out of the project (javascript). 

I will demonstrate using Oxygen to forward lat long pairs to the Overpass API then use Distances.jl to sort bathrooms based on distance. Progress on this work will occur in #10. @michaelfortunato will then help to figure out how we can generate URLs that open apps (google maps). The important part is not just opening the app, but also having it immediately start directing the user. I think its possible but I haven't looked into it

**Reply by acoh64** [reply]
very nice

**Comment by michaelfortunato** [comment]
We'll implement both options. Right now going with Julia Oxygen. I worry about the name "Oxygen" given its letter casing. This could impact kernel development. @acoh64 lets get a meeting on the books to discuss a new liquid neural network program I have been reading about. My team is primariliy focused on the lexer for this project but we want to know how you all are approaching this. Thanks.

**Reply by acoh64** [reply]
what is liquid neural network?
